### PR TITLE
Sims and some minor refactors

### PR DESCRIPTION
**Pull request by maybenikhil**
All commits should be individually tested to make sure they do not break any functions. Goals of this are primarily cleanup + refactors

### CODE REVIEW

File: westcoast/src/main/java/frc/robot/definition/Controllers.java [hunk 1]
```diff
@@ -19,4 +20,35 @@
 
     @NotNull
     PIDController stayOnTarget();
+
+    @NotNull
+    SparkMaxController shooterController();
+
+    final class SparkMaxController {
```
Comment (spacetug) [inline westcoast/src/main/java/frc/robot/definition/Controllers.java]: let's add `static` here
```suggestion
    public static final class SparkMaxController
```

let's call it `ControllerSetter` or something as it is not holding the controller itself.

My recommendation is to define an interface with a single method and creating an anonymous class specifically for the entry:
```java
// here:
public interface ControllerConfigurator {
    void configure(@NotNull SparkMaxPIDController controller)
}

// Variant:
                    @NotNull
                    @Override
                    public ControllerConfigurator shooterController() {
                        return ()-> {
                              controller.setP(0.00015);
                              controller.setI(0.0);
                              controller.setIZone(0.0);
                              controller.setD(0.0007);
                              controller.setFF(0.0002);
                              controller.setOutputRange(0.0, 1.0);
                        };
                    }
```

or even better, we just set it in the variant:
```java
                    @NotNull
                    @Override
                    public void shooterConfig(@NotNull SparkMaxPIDController controller) {
                              controller.setP(0.00015);
                              controller.setI(0.0);
                              controller.setIZone(0.0);
                              controller.setD(0.0007);
                              controller.setFF(0.0002);
                              controller.setOutputRange(0.0, 1.0);
                    }
```

I'm inclined to say the last option is the one we should go

File: westcoast/src/main/java/frc/commands/drive/LeaveLine.java [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
 
     public LeaveLine(
-            double encoderTicks,
+            double distance,
             @NotNull ProfiledPIDController left,
             @NotNull ProfiledPIDController right,
             @NotNull DriveTrainSubsystem drive
     ) {
-        this.encoderTicks = encoderTicks;
+        this.encoderTicks = distance * 64.0;
```
Comment (spacetug) [inline westcoast/src/main/java/frc/commands/drive/LeaveLine.java:27]: if we keep using ticks here why not just accept in encoder tick units whatever a tick is?
I would say if you really think that is a must have necessity (I don't see as such for a few reasons) we should keep accepting ticks in the constructor and have a function somewhere else that make the conversion.

File: westcoast/simgui-ds.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+{
```
Comment (spacetug) [inline westcoast/simgui-ds.json]: I think you can add the sim jsons to the gitignore

File: westcoast/build.gradle [hunk 4]
```diff
@@ -12,6 +12,7 @@ sourceCompatibility = javaVersion
 targetCompatibility = javaVersion
 
 def ROBOT_MAIN_CLASS = "frc.robot.Main"
+def includeDesktopSupport = true
```
Comment (spacetug) [inline westcoast/build.gradle]: this is doing nothing the sim is already available

**Review by spacetug — CHANGES_REQUESTED** [review]
I think you should make a PR for [Change Color to be DS based](https://github.com/team4186/season-2022/pull/1/commits/9b1ec729b33d072aab2dc54e60956e1b021a05de)
And remove [Change Color to FMS Based](https://github.com/team4186/season-2022/pull/1/commits/08ea6b33f6ae346dda9f90d9e9c103087c99d2fe) as change color using DS is a better implementation. also [added simulation support](https://github.com/team4186/season-2022/pull/1/commits/dd4abc6c541ebd0eec8059e2f19a65a8dd631ac5) is useless, the sim is already available in the project. I've used it a little. Clean up the json the sim uses or put the files in the gitignore rules

File: westcoast/src/main/java/frc/robot/definition/Controllers.java (hunk 1 above)
Comment (maybenikhil) [reply westcoast/src/main/java/frc/robot/definition/Controllers.java]: Last option is the most elegant, will do

File: westcoast/src/main/java/frc/commands/drive/LeaveLine.java (hunk 2 above)
Comment (maybenikhil) [reply westcoast/src/main/java/frc/commands/drive/LeaveLine.java:27]: I am going to fix this natively over the weekend (get distance will return in meters [wpilib's preferred unit]) so we can then tune stuff using their tuners.

**Comment by maybenikhil** [comment]
Because i suck at git i'm gonna keep the color changes in this PR, but going forwards i'll try to do more PRs instead of just having one

File: westcoast/src/main/java/frc/robot/definition/Controllers.java (hunk 1 above)
Comment (maybenikhil) [reply westcoast/src/main/java/frc/robot/definition/Controllers.java]: Last option is elegant, but i'm not sure how to effectively implement so I am giving up and doing it the 2nd way

**Comment by maybenikhil** [comment]
Summary of changes and @TODO:
- Combined FMS/DS changes into one commit for clarity
- Removed simulation stuff (added it to gitignore)

Todo:
- Change fundamental backbone of leaveline to run in meters (tune accordingly)
- Figure out how to streamline controller configuration
- Test all changes thoroughly!

**Comment by maybenikhil** [comment]
Closed, see #2, #3, #4, and #5

**Pull request closed without merging**